app/Console/Commands/SupportEvents.php

```diff
@@ -0,0 +1,197 @@
+            $existing = Event::where('vatsim_id', $event->id)->first();
+            if ($existing !== null) {
+                $this->info("Skipping already processed event ".$event->id);
+                continue;
+            } // this event has already been processed
+
+            $pull_this_event = false;
+            $organizer = null;
+
+            // parse times
+            $start_time = Carbon::parse($event->start_time);
+            $end_time = Carbon::parse($event->end_time);
+
+            foreach ($event->airports as $airport) {
+                if (in_array($airport->icao, $all_events_lut)) {
+                    $pull_this_event = true;
+                    $organizer = $inverse_lut[$airport->icao];
+                    break;
+                }
+                if (in_array($airport->icao, $fno_or_live_only_lut)) {
+                    // is it a live event?
+                    if (str_contains(strtolower($event->name), 'live')) {
+                        // live event - pull this event too
+                        $pull_this_event = true;
+                        $organizer = $inverse_lut[$airport->icao];
+                        break;
+                    }
+                    // is it on a friday? (FNO)
+                    if ($start_time->dayOfWeek == 6) { // 06: Friday
+                        // FNO - pull this event too
+                        $pull_this_event = true;
+                        $organizer = $inverse_lut[$airport->icao];
+                        break;
+                    }
+                }
+            }
+
+            if (!$pull_this_event) {
+                continue;
+            }
+
+            $this->info('Creating support event with vatsim id '.$event->id);
+
+            // download the event banner
+            // public/storage/event_banners/vatsim_ID.ext
+
+            $this->info($event->id.': Downloading banner');
+
+            $public_url = '/event_banners/vatsim_'.$event->id.substr($event->banner, -4);
+
+            Storage::disk('public')->put($public_url, file_get_contents($event->banner));
+
+            // create the event in our database
+
+            $this->info($event->id.': Saving to database');
+
+            $our_event = new Event;
+            $our_event->name = $event->name;
+            $our_event->host = $organizer;
+            $our_event->description = $event->description;
+            $our_event->date = $start_time->format("m/d/Y");
+            $our_event->start_time = $start_time->toTimeString('minute');
+            $our_event->end_time = $end_time->toTimeString('minute');
+
+            $our_event->banner_path = $public_url;
+            $our_event->reduceEventBanner();
+
+            $our_event->banner_path = '/storage/'.$public_url;
+            $our_event->status = 0;
+            $our_event->reg = 0;
+            $our_event->type = 2; // auto - unverified
+            $our_event->vatsim_id = $event->id;
+            $our_event->save();
```

Comment:
Let's name this something other than `our_event`. On first glance I thought this was referencing an event that we (ZTL) have scheduled

Reply:
Changed to `new_event`, marking as resolved.